controllers/common.go

```diff
@@ -22,13 +23,43 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	workflowNameSuffix     = "-workflow-counter"
+	jobNameStepInfix       = "-workflow-step-"
+	logDirNameInfix        = "-workflow-step-"
+	workflowStepNumInvalid = -1
+)
+
 type Reconciler struct {
 	Client  client.Client
 	Kclient kubernetes.Interface
 	Log     logr.Logger
 	Scheme  *runtime.Scheme
 }
 
+func (r *Reconciler) GetJobName(instance interface{}, workflowStepNum int) string {
+	if typedInstance, ok := instance.(*v1beta1.Tobiko); ok {
+		if len(typedInstance.Spec.Workflow) == 0 || workflowStepNum == workflowStepNumInvalid {
+			return typedInstance.Name
+		} else {
+			workflowStepName := typedInstance.Spec.Workflow[workflowStepNum].StepName
+			return typedInstance.Name + "-" + workflowStepName + jobNameStepInfix + strconv.Itoa(workflowStepNum)
+		}
+	} else if typedInstance, ok := instance.(*v1beta1.Tempest); ok {
+		return typedInstance.Name
+	} else {
+		return ""
+	}
+}
+
+func (r *Reconciler) GetWorkflowConfigMapName(instance client.Object) string {
+	return instance.GetName() + workflowNameSuffix
+}
+
+func (r *Reconciler) GetLogDirName(frameworkName string, instance client.Object, workflowStepNum int) string {
+	return frameworkName + "-" + instance.GetName() + logDirNameInfix + strconv.Itoa(workflowStepNum)
```

Comment:
IIUC, the log dir name could be something like this:
"<someprefix>-<workflowstepname>-workflow-step-<workflowStepNum>-workflow-step-<workflowStepNum>"

Aren't the last elements redundant?